jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -52,33 +53,29 @@
 @ManagedObject("JSR356 Client Container")
 public class JavaxWebSocketClientContainer extends JavaxWebSocketContainer implements javax.websocket.WebSocketContainer
 {
-    protected final WebSocketCoreClient coreClient;
+    protected WebSocketCoreClient coreClient;
+    protected Supplier<WebSocketCoreClient> coreClientFactory;
     private final JavaxWebSocketClientFrameHandlerFactory frameHandlerFactory;
-    private ClassLoader contextClassLoader;
     private DecoratedObjectFactory objectFactory;
     private WebSocketExtensionRegistry extensionRegistry;
 
     public JavaxWebSocketClientContainer()
     {
-        this(new WebSocketCoreClient());
-        this.coreClient.getHttpClient().setName("Javax-WebSocketClient@" + Integer.toHexString(this.coreClient.getHttpClient().hashCode()));
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this(() -> new WebSocketCoreClient());
```

Comment:
Please set the HttpClient name here to `Javax-WebSocketClient@` + hexHashCode.
This helps in troubleshooting immensely.

Reply:
This Supplier should addBean or addManaged to JavaxWebSocketClientContainer.

Reply:
Good idea to make naming a factory responsibility.  However, I don't think the factory should be responsible for management of the things it creates.